staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go

```diff
@@ -138,7 +140,9 @@ func TestTimeouts(t *testing.T) {
 
 				service, err = NewGRPCService(socketName.endpoint, tt.callTimeout)
 				if err != nil {
-					t.Fatalf("failed to create envelope service, error: %v", err)
+					kubeAPIServerErrc <- fmt.Errorf("failed to create envelope service, error: %v", err)
+					kubeAPIServerWG.Done()
```

Comment:
This test looks very precariously arranged, I don't think I can lgtm this without thinking really hard. Can you fix this test in a separate PR?

I think I'd prefer to change the shutdown mechanism over multiple .Done() calls, this is really hard to understand.

Reply:
Agree! Removed it from this PR and will fix it in a follow-up.